### PR TITLE
Add TSAssertsKeyword support

### PR DESCRIPTION
**Pull request by torifat**
Add support for `TSAssertsKeyword` mentioned in #12 

[TSAssertsKeyword](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions)

**Use cases:**

```
(value: unknown): value is string => true;

(value: unknown): asserts value => {};

(value: unknown): asserts value is string => {};
```

### CODE REVIEW

File: packages/@romejs/js-parser/parser/typescript.ts [hunk 1]
```diff
@@ -1327,24 +1337,44 @@ export function parseTSTypeOrTypePredicateAnnotation(
...
+  let type;
+  if (typePredicateVariable) {
+    type = parseTSTypeAnnotation(parser, /* eatColon */ false);
+  }
 
-  const typePredicateStart = parser.getLoc(typePredicateVariable).start;
+  const typePredicateStart = parser.getLoc(
+    assertsModifier || typePredicateVariable,
+  ).start;
   parser.popScope('TYPE');
```
Comment (torifat) [inline packages/@romejs/js-parser/parser/typescript.ts]: Do you think it's better to split this into multiple functions?

File: packages/@romejs/js-ast/typescript/TSTypePredicate.ts [hunk 2]
```diff
@@ -5,20 +5,28 @@
...
+} from '../index';
 import {createBuilder} from '../utils';
 
 export type TSTypePredicate = JSNodeBase & {
   type: 'TSTypePredicate';
   parameterName: Identifier | TSThisType;
-  typeAnnotation: AnyPrimaryType;
+  typeAnnotation?: AnyPrimaryType;
+  assertsModifier?: TSAssertsKeyword;
 };
```
Comment (torifat) [inline packages/@romejs/js-ast/typescript/TSTypePredicate.ts]: Do you think it better to split it into union types or this is fine?

**Comment by sebmck** [comment]
Thank you for doing this! Is there value in having the `assertsModifier` be a node instead of just a boolean property called `asserts`?

**Comment by torifat** [comment]
I don't think there's any additional value. I was just mimicking TypeScript AST.

**Comment by sebmck** [comment]
Ah cool. Could you update it to just be a boolean instead? The advantage of having a dedicated node is that you have location information for exactly where the `asserts` keyword location is, but we don't need that.

**Comment by torifat** [comment]
I made those changes but having some trouble with the `start` position. I started with `parser.eatContextual('asserts')` but then `start` will be from `1` but I think it should be from `2`.
```
(value: unknown): asserts value => {};
                1 2
```
Can use `parser.isContextual` and get `position` after that but it seems like a lot of work. Any pointers?

**Comment by sebmck** [comment]
`TSTypePredicate` starting from 1 seems fine. I wouldn't worry too much about it.

**Comment by torifat** [comment]
`hub sync` blew up my branch :'(

**Comment by torifat** [comment]
Master is currently red too.

**Comment by sebmck** [comment]
Thank you so much! This is amazing.

**Pull request merged**